server/ocsp.go

```diff
@@ -334,6 +334,16 @@ func (srv *Server) NewOCSPMonitor(config *tlsConfigKind) (*tls.Config, *OCSPMoni
 	// NOTE: Currently OCSP Stapling is enabled only for the first certificate found.
 	var mon *OCSPMonitor
 	for _, cert := range tc.Certificates {
+		if cert.Leaf == nil {
```

Comment:
I'd appreciate a comment here noting the anticipated scenarios where this applies, so that future maintainers know.  "This is normally non-nil, but can still be nil here when in tests or in some embedded scenarios."

There's just too many twisty corner-cases in TLS handling for it to be reasonable to expect all future maintainers to know which logic branches matter when, or "why we bother with this".

Reply:
Added